System/Random.hs

```diff
@@ -1083,6 +1085,22 @@ uniformIntegerM (l, h) gen
         let v' = v * b + fromIntegral x
         v' `seq` f (mag * b) v'
 
+unbiasedIntMult32 :: MonadRandom g m => Word32 -> g -> m Word32
+unbiasedIntMult32 s g
+  | s == 0    = pure 0
```

Comment:
`s` is never equal to `0`: `uniformR` already handles the case `b == t`. When implementing the uniform range method for `Integer`, I decided to document `s > 0` as a precondition instead of having the redundant check. Then again, the check is probably pretty cheap. I just thought I'd mention this.

Reply:
In that case we can remove the `b == t` check, which would make this `s == 0` not redundant

Reply:
Thanks - I know but hadn't got round to fixing it which is why I wanted to mark this branch as "DO NOT MERGE" - thanks very much for spotting it :)

Reply:
I just saw the other change below: `r = s + 1`. Which means with `s == maxBound` the outcome will be division by zero exception, not the `s == 0`

Reply:
Yes I know and I already checked that it's broken for `maxBound` - thanks for spotting it though :)

I promise I will fix it tomorrow.